### PR TITLE
Generate an ID for each behavior that is added.

### PR DESCRIPTION
**Pull request by cyli**
Part of #271.

In the interest of smaller PRs, this PR only returns the UUID when the behavior is created and does not support listing or deleting by ID.

But it is actually not that small, because I also snuck in a conversion of `behaviors.py` from `characteristic` to `attrs` (since otherwise this would be like 5 lines).

### CODE REVIEW

**Comment by cyli** [comment]
Marking as RFR since most of the tox envs have passed, just waiting on mac and pypy.

File: mimic/rest/nova_api.py [hunk 1]
```diff
@@ -151,14 +151,19 @@ def register_creation_behavior(self, request, tenant_id):
                     "message": "Stuff is broken, what"
                 }
             }
+
+        The response looks like:
```
Comment (glyph) [inline mimic/rest/nova_api.py]: I think you're missing a colon here.

File: mimic/test/test_nova.py [hunk 2]
```diff
@@ -1374,12 +1374,13 @@ def use_creation_behavior(self, name, parameters, criteria):
...
-                               json.dumps(criterion))
-        set_criteria_response = self.successResultOf(set_criteria)
-        self.assertEqual(set_criteria_response.code, 201)
+        set_criteria = json_request(self, self.root, "POST",
+                                    self.nova_control_endpoint +
+                                    "/behaviors/creation/",
+                                    json.dumps(criterion))
+        response, body = self.successResultOf(set_criteria)
+        self.assertEqual(response.code, 201)
+        self.assertTrue(body.get("id"))
```
Comment (glyph) [inline mimic/test/test_nova.py]: `assertTrue` is just a *tad* too weak of an assertion for the behavior that the docstring of the method alleges to add.  I'd do something like this:

```python
behavior_id = body.get("id")
self.assertIsInstance(behavior_id, text_type)
parsed = uuid.UUID(hex=behavior_id)
self.assertEquals(parsed.version, 4)
```

I also assume clients are going to want to do something with the behavior ID, although it's fine to leave that for a future PR.

**Comment by glyph** [comment]
Aside from the one minor test issue I think this is fine.

File: mimic/test/test_nova.py (hunk 2 above)
Comment (cyli) [inline mimic/test/test_nova.py]: Yep, adding support for deleting behaviors in a future PR.  Would also be good to list at some point.  Will make the test change, thanks!

**Pull request merged**